src/equations/compressible_euler_quasi_1d.jl

```diff
@@ -0,0 +1,371 @@
+* Trixi.jl's visualization tools will visualize the nozzle width by default.
+"""
+struct CompressibleEulerEquationsQuasi1D{RealT <: Real} <:
+       Trixi.AbstractCompressibleEulerEquations{1, 4}
+    gamma::RealT               # ratio of specific heats
+    inv_gamma_minus_one::RealT # = inv(gamma - 1); can be used to write slow divisions as fast multiplications
+
+    function CompressibleEulerEquationsQuasi1D(gamma)
+        γ, inv_gamma_minus_one = promote(gamma, inv(gamma - 1))
+        new{typeof(γ)}(γ, inv_gamma_minus_one)
+    end
+end
+
+have_nonconservative_terms(::CompressibleEulerEquationsQuasi1D) = Trixi.True()
+function varnames(::typeof(cons2cons), ::CompressibleEulerEquationsQuasi1D)
+    ("a_rho", "a_rho_v1", "a_e", "a")
+end
+function varnames(::typeof(cons2prim), ::CompressibleEulerEquationsQuasi1D)
+    ("rho", "v1", "p", "a")
+end
+
+"""
+    initial_condition_convergence_test(x, t, equations::CompressibleEulerEquationsQuasi1D)
+
+A smooth initial condition used for convergence tests in combination with
+[`source_terms_convergence_test`](@ref)
+(and [`BoundaryConditionDirichlet(initial_condition_convergence_test)`](@ref) in non-periodic domains).
+"""
+function initial_condition_convergence_test(x, t,
+                                            equations::CompressibleEulerEquationsQuasi1D)
+    c = 2
+    A = 0.1
+    L = 2
+    f = 1 / L
+    ω = 2 * pi * f
+    ini = c + A * sin(ω * (x[1] - t))
+
+    rho = ini
+    v1 = 1.0
+    e = ini^2 / rho
+    p = (equations.gamma - 1) * (e - 0.5 * rho * v1^2)
+    a = 1.5 - 0.5 * cos(x[1] * pi)
+
+    return prim2cons(SVector(rho, v1, p, a), equations)
+end
+
+"""
+    source_terms_convergence_test(u, x, t, equations::CompressibleEulerEquationsQuasi1D)
+
+Source terms used for convergence tests in combination with
+[`initial_condition_convergence_test`](@ref)
+(and [`BoundaryConditionDirichlet(initial_condition_convergence_test)`](@ref) in non-periodic domains).
+
+This manufactured solution source term is specifically designed for the mozzle width 'a(x) = 1.5 - 0.5 * cos(x[1] * pi)'
+as defined in [`initial_condition_convergence_test`](@ref).
+"""
+@inline function source_terms_convergence_test(u, x, t,
+                                               equations::CompressibleEulerEquationsQuasi1D)
+    # Same settings as in `initial_condition_convergence_test`. 
+    #Derivatives calculated with ForwardDiff.jl
+    c = 2
+    A = 0.1
+    L = 2
+    f = 1 / L
+    ω = 2 * pi * f
+    x1, = x
+    ini(x1, t) = c + A * sin(ω * (x1 - t))
+
+    rho(x1, t) = ini(x1, t)
+    v1(x1, t) = 1.0
+    e(x1, t) = ini(x1, t)^2 / rho(x1, t)
+    p1(x1, t) = (equations.gamma - 1) * (e(x1, t) - 0.5 * rho(x1, t) * v1(x1, t)^2)
+    a(x1, t) = 1.5 - 0.5 * cos(x1 * pi)
+
+    arho(x1, t) = a(x1, t) * rho(x1, t)
+    arhou(x1, t) = arho(x1, t) * v1(x1, t)
+    aE(x1, t) = a(x1, t) * e(x1, t)
+
+    darho_dt(x1, t) = ForwardDiff.derivative(t -> arho(x1, t), t)
```

Comment:
For the other manufactured solutions used in Trixi we do the derivatives manually. 
While this is an elegant solution it also introduces dependency on upstream packages which caused failing tests in the past.
How do @ranocha @sloede feel about this?

Reply:
ForwardDiff is already a dependency of Trixi.jl, no?

Reply:
https://github.com/trixi-framework/Trixi.jl/blob/5c08cf4a4a7407d3ffa90182baafaf8c286b8c33/Project.toml#L12

Reply:
Sorry, I meant "introduce dependency for the tests". As of now, `ForwardDiff` is not required for the tests (apart from the Jacobian computations).

Reply:
Ah, I see. Good point. However, we are using ForwardDiff in tests too? 

https://github.com/jlchan/Trixi.jl/blob/4ca49a570998defa5a1ff0a8650da4e3f077c19a/test/test_special_elixirs.jl#L310

Reply:
Yes - but not yet for the conventional elixir tests but only for more specialized things.

Anyway, I just wanted to point this out - there are no real objections from my side in using the automated derivative here if others are fine with this.